sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -442,6 +442,10 @@ class SessionCatalog(
     name.database.isEmpty && tempTables.contains(formatTableName(name.table))
   }
 
+  def listTemporaryTableOutput(name: String): Seq[Attribute] = {
```

Comment:
Ah. I remember that why I do this in this way. Basically, there are two barriers to reach `getTableMetadata`. Before making change, let me describe here.

1. Redirecting: `listColumns(table)`  -> `listColumns(currentDatabase, tableName)`
2. Table existence failure: `requireTableExists(dbName, tableName)` in `listColumns(currentDatabase, tableName)`.

Anyway, I'm trying to change the above barriers.